cmd/qliksense/root.go

```diff
@@ -95,29 +97,114 @@ func installPorter() (string, error) {
 			}
 		}
 	}
+
 	if downloadPorter {
 		os.Mkdir(qlikSenseHome, os.ModePerm)
 		destination = filepath.Join(qlikSenseHome, porterRuntime)
-		if err = downloadFile(porterURLBase+"/"+porterPermaLink+"/porter-linux-amd64", destination); err != nil {
-			return "", err
-		}
-		os.Chmod(destination, 0755)
-		if runtime.GOOS == "linux" && runtime.GOARCH == "amd64" {
+		// construct url to download porter from
+		porterDownloadURL = constructPorterURL(runtime.GOOS)
+
+		// if runtime.GOOS == "linux" && runtime.GOARCH == "amd64" {
+		if (runtime.GOOS == "linux" && runtime.GOARCH == "amd64") || runtime.GOOS == "darwin" {
+			if err = downloadFile(porterDownloadURL, destination); err != nil {
+				return "", err
+			}
+			os.Chmod(destination, 0755)
 			if _, err = copy(filepath.Join(qlikSenseHome, porterRuntime), porterExe); err != nil {
 				return "", err
 			}
 			os.Chmod(porterExe, 0755)
-		} else {
-			if runtime.GOOS == "windows" {
-				ext = ".exe"
-			}
-			if err = downloadFile(porterURLBase+"/"+porterPermaLink+"/"+"porter-"+runtime.GOOS+"-"+runtime.GOARCH+ext, porterExe); err != nil {
+		} else if runtime.GOOS == "windows" {
+			if err = downloadFile(porterDownloadURL, porterExe); err != nil {
 				return "", err
 			}
 			os.Chmod(porterExe, 0755)
 		}
 	}
+	return porterExe, nil
+}
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Print the version number of qliksense cli",
+	Long:  `All software has versions. This is Hugo's`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Printf("%s (%s, %s)\n", pkg.Version, pkg.Commit, pkg.CommitDate)
+	},
+}
 
+func constructPorterURL(runtimeOS string) string {
+	// FYI: Porter does not support other architectures other than amd64
+	const (
+		porterURLBase = "https://cdn.deislabs.io/porter/"
```

Comment:
same variable is declared above as well ??

Reply:
fixed